src/scheduler.c

```diff
@@ -118,29 +120,35 @@ void *scheduler_thread(void *args)
 
     call_firewall( firewall_cmd, NULL );
 
-    while (1) {
-        struct timespec tm;
+    while( true ) {
```

Comment:
If you want to wait for the clock to start, make it a function so it's clear that's what you're doing.  Also ensure that you sleep for some amount of time.